### PR TITLE
Check for nil infra ref in azurejson controllers

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**: Fix a nil pointer error observed in https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1549613668864364544/artifacts/clusters/bootstrap/controllers/capz-controller-manager/capz-controller-manager-65c9d7b84d-mz2j2/manager.log and https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1549794864755904512/artifacts/clusters/bootstrap/controllers/capz-controller-manager/capz-controller-manager-75fc64f756-mhfkn/manager.log

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Not 100% sure if this is a new bug or if it started being visible in e2e logs because of changes to parallelism.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Check for infra ref nil pointer in azurejson controllers
```

### CODE REVIEW

File: controllers/azurejson_machine_controller.go [hunk 1]
```diff
@@ -147,6 +147,10 @@ func (r *AzureJSONMachineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	_, kind := infrav1.GroupVersion.WithKind("AzureCluster").ToAPIVersionAndKind()
 
 	// only look at azure clusters
+	if cluster.Spec.InfrastructureRef == nil {
```
Comment (mboersma) [inline controllers/azurejson_machine_controller.go:150]: Should we add a test case for `InfrastructureRef == nil` to `TestAzureJSONMachineReconciler`?

Comment (CecileRobertMichon) [reply controllers/azurejson_machine_controller.go:150]: thanks for the nudge, added

**Comment by mboersma** [comment]
/retest

IPv6 provisioning flake, doesn't appear related to this nil check.
